grails-app/services/org/pih/warehouse/core/IdentifierService.groovy

```diff
@@ -166,7 +166,34 @@ class IdentifierService {
         else if (identifier.length() > maxSize) {
             identifier = identifier.substring(0, maxSize)
         }
-        return identifier.toUpperCase()
+
+        // Checking for a duplicates
+        if (validateOrganizationIdentifier(identifier.toUpperCase())) {
+            return identifier.toUpperCase()
+        }
+
+        if (Character.isDigit(identifier.charAt(0))) {
```

Comment:
Can you comment this code? From the looks of it we're adding a prefix to the organization code. 

So let's say we had an organization with code ABC and we tried to add a second one with the same identifier, the second one would be 0ABC? Is that correct? 

Are there requirements in the ticket for this? 

I think I remember Kelsey saying offhand something about appending a number to the code but not a prefix. To me, it seems like we'd want to go ABC, ABC2, ABC3.